### ISSUE TITLE
[ResourceManager] Reenable the ValidateResourceInRestApi test

### ISSUE THREAD

**Issue by fengzhou-msft** [Azure/azure-sdk-for-net#34569]
The recording files for [ValidateResourceInRestApi](https://github.com/Azure/azure-sdk-for-net/blob/59fcdfcdc4d91779bc030425b787a43e38f4f365/sdk/resourcemanager/Azure.ResourceManager/tests/Scenario/SubscriptionOperationsTests.cs#LL186C27-L186C52) is too big and a re-recording of it caused github not able to trigger CI (even an [azp run comment](https://github.com/Azure/azure-sdk-for-net/pull/33147#issuecomment-1364025879) does not work). It was ignored to unblock PR(#33857) merge.

We should reenable the test and update the recordings. If a direct re-recording still doesn't work in a single PR, we may consider to use one PR to remove the recordings and another PR to add back new recordings.
